src/tooltip/ToolTip2.tsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import {
+  Modal,
+  View,
+  I18nManager,
+  Pressable,
+  ViewStyle,
+  FlexStyle,
+  StyleProp,
+  StyleSheet,
+  ColorValue,
+} from 'react-native';
+import { Animated, Text, SafeAreaView } from 'react-native';
+import Triangle from './Triangle';
+import { withTheme } from '../config';
+import { ScreenWidth, ScreenHeight } from '../helpers';
+
+export type ToolTip2Props = {
+  variant?: 'grow' | 'fade' | 'zoom';
+  withPointer?: boolean;
+  title?: string;
+  popover?: React.ReactElement<{}>;
+  toggleOnPress?: boolean;
+  toggleAction?: 'onPress' | 'onLongPress';
+  width?: FlexStyle['width'];
+  containerStyle?: StyleProp<ViewStyle>;
+  pointerColor?: ColorValue;
+  onClose?(): void;
+  onOpen?(): void;
+  open?: boolean;
+  overlayColor?: ColorValue;
+  backgroundColor?: ColorValue;
+  closeOnlyOnBackdropPress?: boolean;
+};
+
+const ToolTip2: React.FunctionComponent<ToolTip2Props> = ({
+  children,
+  title,
+  popover,
+  withPointer = true,
+  overlayColor,
+  backgroundColor = '#616161',
+  pointerColor,
+  variant = 'grow',
+  toggleAction = 'onPress',
+  onOpen = () => {},
+  onClose = () => {},
+  open: propOpen = false,
+  width: propWidth = 200,
+  //   closeOnlyOnBackdropPress,
+  containerStyle,
+}) => {
+  const { current: animation } = React.useRef<Animated.Value>(
+    new Animated.Value(0)
+  );
+
+  let element = React.useRef<View>(null);
+  const [open, setOpen] = React.useState(false);
+  const [containerDimentions, setContainerDimentions] = React.useState({
+    height: 0,
+    px: 0,
+    py: 0,
+    width: 0,
+  });
+  const [tooltipDimention, setTooltipDimention] = React.useState({
+    height: 0,
+  });
+
+  const toogleToolTip = () => {
+    setOpen(!open);
+    if (open) {
+      onClose();
+    } else {
+      onOpen();
+    }
+  };
+
+  React.useEffect(() => {
+    Animated.timing(animation, {
+      toValue: Number(open),
+      useNativeDriver: true,
+      duration: 140,
+    }).start();
+  }, [open, animation]);
+
+  React.useEffect(() => {
+    requestAnimationFrame(
+      () =>
+        element.current &&
+        element.current.measure((_x, _y, width, height, px, py) => {
+          setContainerDimentions({ width, height, px, py });
+        })
+    );
+  }, []);
```

Comment:
It's working

Reply:
Are you really sure about it? Because during scrolling components inside don't get remounted and this will be only called once which is first mount and your screen absolute coordinates might be wrong.